Readability.js

```diff
@@ -1722,6 +1722,23 @@ Readability.prototype = {
   isProbablyReaderable: function(helperIsVisible) {
     var nodes = this._getAllNodesWithTag(this._doc, ["p", "pre"]);
 
+    // Get <div> nodes which have <br> node(s) and append them into the `nodes` variable.
+    // Some articles' DOM structures might look like
+    // <div>
+    //   Sentences<br>
+    //   <br>
+    //   Sentences<br>
+    // </div>
+    var brNodes = this._getAllNodesWithTag(this._doc, ["div > br"]);
+    if (brNodes.length > 1) {
+      var set = new Set();
+      [].forEach.call(brNodes, function(node) {
+        var div = node.parentNode;
+        !set.has(div) && set.add(div);
+      });
+      nodes = [].concat.call(Array.from(set), nodes);
+    }
+
```

Comment:
Oh, right, I did miss that. Sorry.

I don't know what the perf impact of this suggestion would be. You'd have to try... it might be difficult because this won't select anything in nodejs, because there's no `querySelectorAll` implementation in JSDOMParser (and I don't think we should write one). It will only work against the real browser and maybe `jsdom`. But we need to keep testing JSDOMParser for Readability because we will use the DOM from there for the actual conversion to reader mode. So this is a bit tricky.

Otherwise, it looks at least like we could try it. You would need to modify the test framework to always use a `jsdom`-generated DOM for the `isProbablyReaderable` call. Then there are perf benchmarks already defined in the package.json that I think it would be possible to use to check what the impact is. If it's 5-10% slower that's probably OK. If it's 3 times as slow as it is now, that's probably less OK. Note that IIRC they might be quite noisy so you might need to run them multiple times.

Finally, note that the 'point' of a Set is that you can just call `.add` without checking `.has` - there will not be duplicates. :-)